### PR TITLE
[API Compatibility No.186] Add target alias support for paddle.nn.functional.mse_loss -part

### PR DESCRIPTION
**Pull request by Aidenwu0209**
### PR Category
User Experience

### PR Types
New features

### Description
- Add `@param_one_alias(["label", "target"])` on `paddle.nn.functional.mse_loss` to support PyTorch-style keyword argument `target` as alias of `label`.
- Update `mse_loss` docstring to document `label` alias: `target`.
- Add dynamic-graph unit test in `test/legacy_test/test_mse_loss.py` to verify `label=` and `target=` produce identical outputs for `reduction="none"` and `reduction="mean"`.

No signature or core logic changes were introduced.

Related task: #76301 (No.186)

### 是否引起精度变化
否

### CODE REVIEW

**Review by zhwesky2010 — DISMISSED** [review]
LGTM

File: test/legacy_test/test_mse_loss.py [hunk 1]
```diff
@@ -352,6 +353,28 @@ def test_dygraph_and_grad(self):
             np.testing.assert_allclose(x.grad.shape, x.shape)
 
 
+class TestNNFunctionalMseLossAlias(unittest.TestCase):
+    def test_target_alias_dygraph(self):
+        paddle.disable_static()
+        try:
```
Comment (zhwesky2010) [inline test/legacy_test/test_mse_loss.py]: 测试代码都不需要用try（除非你是特意测异常抛出的case），注意保证测试结果正确可运行。

Comment (Aidenwu0209) [reply test/legacy_test/test_mse_loss.py]: 已按建议修改：去掉 try/finally，改为 with base.dygraph.guard() 管理动态图上下文，保持原有断言逻辑不变。请帮忙再看下，感谢！

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by YqGe585 — APPROVED** [review]
LGTM for IXUCA
